pytorch_lightning/callbacks/progress/rich_progress.py

```diff
@@ -345,8 +346,13 @@ def on_train_epoch_start(self, trainer, pl_module):
             )
         self.refresh()
 
-    def on_validation_epoch_start(self, trainer, pl_module):
-        if self.total_val_batches > 0:
+    def on_validation_batch_start(
+        self, trainer: "pl.Trainer", pl_module: "pl.LightningModule", batch: Any, batch_idx: int, dataloader_idx: int
+    ) -> None:
+        if self.is_dataloader_changed(dataloader_idx):
+            if self.val_progress_bar_id is not None:
```

Comment:
Should these 2 be an `or`?

If it does not exist OR the dataloader changed, we want to change it

Reply:
the condition checks if it does exist, then remove the old one from the display.

Reply:
Oh I didn't notice the hook change